src/server/report/app.js

```diff
@@ -522,7 +507,53 @@ class App extends LitElement {
 				searchParams.set(name, params[name]);
 			}
 		}
-		page.redirect(`${this._root}?${searchParams.toString()}`);
+		this.#navigate(`${this._root}?${searchParams.toString()}`);
+	}
+	#handleClick(e) {
+		if (e.metaKey || e.ctrlKey || e.shiftKey || e.defaultPrevented) return;
+
+		let el = e.target;
+		const eventPath = e.composedPath();
+		for (let i = 0; i < eventPath.length; i++) {
+			if (eventPath[i].nodeName?.toUpperCase() !== 'A' || !eventPath[i]?.href) continue;
+			el = eventPath[i];
+			break;
+		}
+		if (el?.nodeName.toUpperCase() !== 'A') return;
```

Comment:
This handles all `click` events in the application and if they're a `<a>` does a SPA navigation. It makes some assumptions that all the links are within the SPA, which for this app is true.

Reply:
Essentially this logic takes into account that you can click on a thing deeply nested inside an `<a>` (maybe shadow roots) and we want to find the first `<a>` in that composed path. But then it might not find an `<a>` at all, so we bail.

Reply:
Yep -- your way totally works too and I like that better. Updated.